### ISSUE TITLE
php8

### ISSUE THREAD

**Issue by indrisepos** [auroraextensions/simplereturns#3]
Hello,

I tried to install the module on Magento 2.4.5-p2 with the composer --ignore-platform-reqs parameter. It gave the following errors during upgrade and static deploy. I used Php version 8.1.17.

`Updating modules:
PHP Deprecated:  Optional parameter $meta declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Rma/DataProvider.php on line 65
PHP Deprecated:  Optional parameter $data declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Rma/DataProvider.php on line 65
PHP Deprecated:  Optional parameter $addFieldStrategies declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Rma/DataProvider.php on line 65
PHP Deprecated:  Optional parameter $addFilterStrategies declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Rma/DataProvider.php on line 65
PHP Deprecated:  Optional parameter $meta declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Package/DataProvider.php on line 57
PHP Deprecated:  Optional parameter $data declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Package/DataProvider.php on line 57
PHP Deprecated:  Optional parameter $meta declared before required parameter $searchCriteriaBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Form/Rma/DataProvider.php on line 86
PHP Deprecated:  Optional parameter $data declared before required parameter $searchCriteriaBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Form/Rma/DataProvider.php on line 86
PHP Deprecated:  Optional parameter $components declared before required parameter $urlBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Actions.php on line 49
PHP Deprecated:  Optional parameter $data declared before required parameter $urlBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Actions.php on line 49
`

`PHP Deprecated:  Optional parameter $meta declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Rma/DataProvider.php on line 65
PHP Deprecated:  Optional parameter $data declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Rma/DataProvider.php on line 65
PHP Deprecated:  Optional parameter $addFieldStrategies declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Rma/DataProvider.php on line 65
PHP Deprecated:  Optional parameter $addFilterStrategies declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Rma/DataProvider.php on line 65
PHP Deprecated:  Optional parameter $meta declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Package/DataProvider.php on line 57
PHP Deprecated:  Optional parameter $data declared before required parameter $viewModel is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Grid/Package/DataProvider.php on line 57
PHP Deprecated:  Optional parameter $meta declared before required parameter $searchCriteriaBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Form/Rma/DataProvider.php on line 86
PHP Deprecated:  Optional parameter $data declared before required parameter $searchCriteriaBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Form/Rma/DataProvider.php on line 86
PHP Deprecated:  Optional parameter $components declared before required parameter $urlBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Actions.php on line 49
PHP Deprecated:  Optional parameter $data declared before required parameter $urlBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Actions.php on line 49
Compilation was started.
Repositories code generation... 1/9 [===>------------------------]  11% < 1 sec 90.0 MiBPHP Deprecated:  Optional parameter $data declared before required parameter $packageRepository is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Block/Adminhtml/Label/Generate.php on line 55
PHP Deprecated:  Optional parameter $meta declared before required parameter $searchCriteriaBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Form/Package/DataProvider.php on line 74
PHP Deprecated:  Optional parameter $data declared before required parameter $searchCriteriaBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Form/Package/DataProvider.php on line 74
PHP Deprecated:  Optional parameter $labels declared before required parameter $searchCriteriaBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/DataProvider/Form/Package/DataProvider.php on line 74
PHP Deprecated:  Optional parameter $components declared before required parameter $orderRepository is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Order/IncrementId.php on line 50
PHP Deprecated:  Optional parameter $data declared before required parameter $orderRepository is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Order/IncrementId.php on line 50
PHP Deprecated:  Optional parameter $components declared before required parameter $urlBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Rma.php on line 75
PHP Deprecated:  Optional parameter $data declared before required parameter $urlBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Rma.php on line 75
PHP Deprecated:  Optional parameter $components declared before required parameter $urlBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Action.php on line 69
PHP Deprecated:  Optional parameter $data declared before required parameter $urlBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Action.php on line 69
PHP Deprecated:  Optional parameter $components declared before required parameter $urlBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Order.php on line 75
PHP Deprecated:  Optional parameter $data declared before required parameter $urlBuilder is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Order.php on line 75
PHP Deprecated:  Optional parameter $components declared before required parameter $rmaRepository is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Rma/IncrementId.php on line 54
PHP Deprecated:  Optional parameter $data declared before required parameter $rmaRepository is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Ui/Component/Listing/Column/Rma/IncrementId.php on line 54
PHP Deprecated:  Optional parameter $data declared before required parameter $validator is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Sales/Order/HistoryView.php on line 59
PHP Deprecated:  Optional parameter $data declared before required parameter $validator is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Sales/Order/Info/ActionsView.php on line 58
PHP Deprecated:  Optional parameter $data declared before required parameter $storeManager is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Rma/EditView.php on line 115
PHP Deprecated:  Optional parameter $data declared before required parameter $tokenizer is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Rma/ViewView.php on line 115
PHP Deprecated:  Optional parameter $data declared before required parameter $validator is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Rma/ListView.php on line 64
PHP Deprecated:  Optional parameter $data declared before required parameter $orderAdapter is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Rma/CreateView.php on line 66
PHP Deprecated:  Optional parameter $data declared before required parameter $validator is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Orders/ResultsView.php on line 53
PHP Deprecated:  Optional parameter $data declared before required parameter $simpleReturnRepository is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Package/Form/InfoView.php on line 81
PHP Deprecated:  Optional parameter $data declared before required parameter $simpleReturnRepository is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Package/ViewView.php on line 103
PHP Deprecated:  Optional parameter $data declared before required parameter $moduleConfig is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Package/ListView.php on line 51
PHP Deprecated:  Optional parameter $data declared before required parameter $simpleReturnRepository is implicitly treated as a required parameter in vendor/auroraextensions/simplereturns/Model/ViewModel/Package/CreateView.php on line 76
`

**Comment by nickolasburr** [comment]
@indrisepos This issue is fixed in 1cf0da40 and will be included in the next release.

**Comment by indrisepos** [comment]
Thank you very much, nice work! I tested the latest commit 3874b61ecdb7e5b28074aeb0ce084cdf06656dfc, I only see one issue during setup:di:compile:

`Repositories code generation... 1/9 [===>------------------------]  11% < 1 sec 92.0 MiBPHP Fatal error:  AuroraExtensions\SimpleReturns\Controller\Adminhtml\Package\CreatePost and AuroraExtensions\SimpleReturns\Component\System\ModuleConfigTrait define the same property ($moduleConfig) in the composition of AuroraExtensions\SimpleReturns\Controller\Adminhtml\Package\CreatePost. However, the definition differs and is considered incompatible. Class was composed in vendor/auroraextensions/simplereturns/Controller/Adminhtml/Package/CreatePost.php on line 61`

I have watched the project, I am happy to test it further.

**Comment by nickolasburr** [comment]
Thanks @indrisepos, any user testing is greatly appreciated. For the above issue, it's also on my radar and will receive a fix in the next release. This is the patch for it:

```
diff --git a/Controller/Adminhtml/Package/CreatePost.php b/Controller/Adminhtml/Package/CreatePost.php
index fc7de21..1f1e723 100644
--- a/Controller/Adminhtml/Package/CreatePost.php
+++ b/Controller/Adminhtml/Package/CreatePost.php
@@ -40,6 +40,7 @@ use Magento\Framework\App\Filesystem\DirectoryList;
 use Magento\Framework\App\RequestInterface;
 use Magento\Framework\Controller\Result\Json as ResultJson;
 use Magento\Framework\Controller\Result\JsonFactory as ResultJsonFactory;
+use Magento\Framework\Controller\Result\Redirect;
 use Magento\Framework\Data\Form\FormKey\Validator as FormKeyValidator;
 use Magento\Framework\Escaper;
 use Magento\Framework\Exception\AlreadyExistsException;
@@ -60,6 +61,14 @@ use function Ramsey\Uuid\v4;
  */
 class CreatePost extends Action implements HttpPostActionInterface
 {
+    /**
+     * @var ConfigInterface $moduleConfig
+     * @method ConfigInterface getConfig()
+     * ---
+     * @method Redirect getRedirect()
+     * @method Redirect getRedirectToPath()
+     * @method Redirect getRedirectToUrl()
+     */
     use ModuleConfigTrait, RedirectTrait;
 
     private const PARAM_RMA_ID = 'rma_id';
@@ -80,9 +89,6 @@ class CreatePost extends Action implements HttpPostActionInterface
     /** @var FormKeyValidator $formKeyValidator */
     private $formKeyValidator;
 
-    /** @var ConfigInterface $moduleConfig */
-    private $moduleConfig;
-
     /** @var PackageInterfaceFactory $packageFactory */
     private $packageFactory;
 

```

**Comment by nickolasburr** [comment]
Resolved as of v1.3.0
